src/core/amount.js

```diff
@@ -113,32 +110,45 @@ Amount.is_valid_full = function(j) {
 };
 
 Amount.NaN = function() {
-  var result = new Amount();
-  result._value = new BigNumber(NaN); // should have no effect
+  const result = new Amount();
+  result._value = new Value(NaN)._value; // should have no effect
   return result;                      // but let's be careful
 };
 
 // be sure that _is_native is set properly BEFORE calling _set_value
 Amount.prototype._set_value = function(value, roundingMode) {
```

Comment:
We should be able to get rid of the `_set_value` method by just requiring that a Value instance is passed into the Amount constructor.